git_machete/client.py

```diff
@@ -1483,7 +1483,9 @@ def branch_def_to_str(lb: str, lb_or_rb: str) -> str:
             debug(f"match_log_to_filtered_reflogs({branch})",
                   "branches containing the given SHA in their filtered reflog: \n%s\n" % "\n".join(log_result()))
 
-        for sha in self.__git.spoonfeed_log_shas(branch):
+        branch_full_hash: FullCommitHash = self.__git.get_full_sha(branch)
```

Comment:
Hmmm while you're here... I can see there's both `get_commit_sha_by_revision` and `get_full_sha`... these should be unified as they basically do the same thing 🤔